### PR TITLE
#18775  New integration tests

### PR DESCRIPTION
**Pull request by nollymar**
Integration tests for #18775 . 
_Tests implemented here will fail until `release-5.3.3` gets merged into master_

Additionally, fixes for this issue were included: https://github.com/dotCMS/core/issues/18804

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        relationshipAPI = APILocator.getRelationshipAPI();
+    }
+
+    /**
+     * <b>Method to test:</b> {@link DependencyManager#setDependencies()} <p>
+     * <b>Given Scenario:</b> A bundle contains a content with a self-join relationship <p>
+     * <b>ExpectedResult:</b> The Dependency Manager should include all related content to the bundle
+     */
+    @Test
+    public void test_dependencyManager_shouldIncludeSelfRelationships()
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:70]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java [hunk 2]
```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
...
+                .contentTypeId(blogContentType.id())
+                .values(String.valueOf(RELATIONSHIP_CARDINALITY.MANY_TO_MANY.ordinal()))
+                .relationType(commentContentType.variable()).required(false).build();
+
+        contentTypeFieldAPI.save(field, user);
+
+        //Creates parent content
+        Contentlet blogContentParent = new ContentletDataGen(blogContentType.id())
+                .languageId(APILocator.getLanguageAPI().getDefaultLanguage().getId())
+                .setProperty("title", "blogContentParent")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:271]: Codacy found an issue: [The String literal "blogContentParent" appears 4 times in this file; the first occurrence is on line 271](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java [hunk 3]
```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
...
+                .setProperty("urlTitle", "blogContentParent")
+                .setProperty("author", "systemUser")
+                .setProperty("sysPublishDate", new Date())
+                .setProperty("body", "blogBody").next();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:275]: Codacy found an issue: [The String literal "blogBody" appears 4 times in this file; the first occurrence is on line 275](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java [hunk 4]
```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
...
+                .values(String.valueOf(RELATIONSHIP_CARDINALITY.MANY_TO_MANY.ordinal()))
+                .relationType(commentContentType.variable()).required(false).build();
+
+        contentTypeFieldAPI.save(field, user);
+
+        //Creates parent content
+        Contentlet blogContentParent = new ContentletDataGen(blogContentType.id())
+                .languageId(APILocator.getLanguageAPI().getDefaultLanguage().getId())
+                .setProperty("title", "blogContentParent")
+                .setProperty("urlTitle", "blogContentParent")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:272]: Codacy found an issue: [The String literal "urlTitle" appears 4 times in this file; the first occurrence is on line 272](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java [hunk 5]
```diff
@@ -46,7 +55,9 @@
     private static Structure structure = null;
     private static User user = null;
     private static RelationshipAPI relationshipAPI = null;
+    private static ContentletAPI contentletAPI;
     private static ContentTypeAPI contentTypeAPI = null;
+    private static FieldAPI contentTypeFieldAPI = null;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:60]: Codacy found an issue: [Avoid using redundant field initializer for 'contentTypeFieldAPI'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+        final PushPublisherConfig config = new PushPublisherConfig();
+        final ContentType contentType = getContentTypeWithSelfJoinRelationship();
+        final ContentletDataGen dataGen = new ContentletDataGen(contentType.id());
+
+        //Creates parent content
+        Contentlet blogContentParent = dataGen
+                .languageId(languageId)
+                .setProperty("title", "blogContentParent")
+                .setProperty("urlTitle", "blogContentParent")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:81]: Codacy found an issue: [The String literal "urlTitle" appears 4 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java [hunk 7]
```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
...
+                .relationType(commentContentType.variable()).required(false).build();
+
+        contentTypeFieldAPI.save(field, user);
+
+        //Creates parent content
+        Contentlet blogContentParent = new ContentletDataGen(blogContentType.id())
+                .languageId(APILocator.getLanguageAPI().getDefaultLanguage().getId())
+                .setProperty("title", "blogContentParent")
+                .setProperty("urlTitle", "blogContentParent")
+                .setProperty("author", "systemUser")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:273]: Codacy found an issue: [The String literal "author" appears 4 times in this file; the first occurrence is on line 273](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+                .setProperty("author", "systemUser")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:82]: Codacy found an issue: [The String literal "author" appears 4 times in this file; the first occurrence is on line 82](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+                .setProperty("sysPublishDate", new Date())
+                .setProperty("body", "blogBody").next();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:84]: Codacy found an issue: [The String literal "blogBody" appears 4 times in this file; the first occurrence is on line 84](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+            throws DotSecurityException, DotBundleException, DotDataException {
+
+        final PushPublisherConfig config = new PushPublisherConfig();
+        final ContentType contentType = getContentTypeWithSelfJoinRelationship();
+        final ContentletDataGen dataGen = new ContentletDataGen(contentType.id());
+
+        //Creates parent content
+        Contentlet blogContentParent = dataGen
+                .languageId(languageId)
+                .setProperty("title", "blogContentParent")
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:80]: Codacy found an issue: [The String literal "blogContentParent" appears 4 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java [hunk 11]
```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
...
+
+    /**
+     * <b>Method to test:</b> {@link RelationshipAPI#dbRelatedContent(Relationship, Contentlet)} <p>
+     * <b>Given Scenario:</b> A content is related to another (same content type/self-join relationship) <p>
+     * <b>ExpectedResult:</b> {@link RelationshipAPI#dbRelatedContent(Relationship, Contentlet)} should always return related content
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void testDbRelatedContentWhenSelfJoinRelationship()
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:315]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java [hunk 12]
```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
...
 
+    /**
+     * <b>Method to test:</b> {@link RelationshipAPI#dbRelatedContent(Relationship, Contentlet)} <p>
+     * <b>Given Scenario:</b> A content is related to another (different content types) <p>
+     * <b>ExpectedResult:</b> {@link RelationshipAPI#dbRelatedContent(Relationship, Contentlet)} should always return related content
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void testDbRelatedContentWhenNonSelfJoinRelationship()
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:255]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java [hunk 13]
```diff
@@ -231,6 +244,132 @@ public void testconvertRelationshipToRelationshipField_SelfRelated_Success() thr
...
+
+        contentTypeFieldAPI.save(field, user);
+
+        //Creates parent content
+        Contentlet blogContentParent = new ContentletDataGen(blogContentType.id())
+                .languageId(APILocator.getLanguageAPI().getDefaultLanguage().getId())
+                .setProperty("title", "blogContentParent")
+                .setProperty("urlTitle", "blogContentParent")
+                .setProperty("author", "systemUser")
+                .setProperty("sysPublishDate", new Date())
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:274]: Codacy found an issue: [The String literal "sysPublishDate" appears 4 times in this file; the first occurrence is on line 274](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java [hunk 14]
```diff
@@ -0,0 +1,256 @@
...
+
+    private static long languageId;
+    private static User user;
+    private static BundleAPI bundleAPI;
+    private static ContentletAPI contentletAPI;
+    private static FieldAPI contentTypeFieldAPI;
+    private static RelationshipAPI relationshipAPI;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:50]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java [hunk 15]
```diff
@@ -0,0 +1,256 @@
...
+    /**
+     * <b>Method to test:</b> {@link DependencyManager#setDependencies()} <p>
+     * <b>Given Scenario:</b> A bundle contains a content with a relationship <p>
+     * <b>ExpectedResult:</b> The Dependency Manager should include all related content to the bundle
+     * @throws DotSecurityException
+     * @throws DotBundleException
+     * @throws DotDataException
+     */
+    @Test
+    public void test_dependencyManager_shouldIncludeRelationships()
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:134]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+        final ContentType contentType = getContentTypeWithSelfJoinRelationship();
+        final ContentletDataGen dataGen = new ContentletDataGen(contentType.id());
+
+        //Creates parent content
+        Contentlet blogContentParent = dataGen
+                .languageId(languageId)
+                .setProperty("title", "blogContentParent")
+                .setProperty("urlTitle", "blogContentParent")
+                .setProperty("author", "systemUser")
+                .setProperty("sysPublishDate", new Date())
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:83]: Codacy found an issue: [The String literal "sysPublishDate" appears 4 times in this file; the first occurrence is on line 83](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

**Review by erickgonzalez — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:82]: Codacy found an issue: [The String literal "author" appears 4 times in this file; the first occurrence is on line 82](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:60]: Codacy found an issue: [Avoid using redundant field initializer for 'contentTypeFieldAPI'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:275]: Codacy found an issue: [The String literal "blogBody" appears 4 times in this file; the first occurrence is on line 275](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:272]: Codacy found an issue: [The String literal "urlTitle" appears 4 times in this file; the first occurrence is on line 272](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:70]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:80]: Codacy found an issue: [The String literal "blogContentParent" appears 4 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:274]: Codacy found an issue: [The String literal "sysPublishDate" appears 4 times in this file; the first occurrence is on line 274](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:50]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 15 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:134]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:315]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:271]: Codacy found an issue: [The String literal "blogContentParent" appears 4 times in this file; the first occurrence is on line 271](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:255]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:83]: Codacy found an issue: [The String literal "sysPublishDate" appears 4 times in this file; the first occurrence is on line 83](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:273]: Codacy found an issue: [The String literal "author" appears 4 times in this file; the first occurrence is on line 273](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:81]: Codacy found an issue: [The String literal "urlTitle" appears 4 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:84]: Codacy found an issue: [The String literal "blogBody" appears 4 times in this file; the first occurrence is on line 84](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:271]: Codacy found an issue: [The String literal "blogContentParent" appears 4 times in this file; the first occurrence is on line 271](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 15 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:134]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:50]: Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:255]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:273]: Codacy found an issue: [The String literal "author" appears 4 times in this file; the first occurrence is on line 273](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:84]: Codacy found an issue: [The String literal "blogBody" appears 4 times in this file; the first occurrence is on line 84](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:275]: Codacy found an issue: [The String literal "blogBody" appears 4 times in this file; the first occurrence is on line 275](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:80]: Codacy found an issue: [The String literal "blogContentParent" appears 4 times in this file; the first occurrence is on line 80](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:83]: Codacy found an issue: [The String literal "sysPublishDate" appears 4 times in this file; the first occurrence is on line 83](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:272]: Codacy found an issue: [The String literal "urlTitle" appears 4 times in this file; the first occurrence is on line 272](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:82]: Codacy found an issue: [The String literal "author" appears 4 times in this file; the first occurrence is on line 82](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:274]: Codacy found an issue: [The String literal "sysPublishDate" appears 4 times in this file; the first occurrence is on line 274](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:81]: Codacy found an issue: [The String literal "urlTitle" appears 4 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:315]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java:70]: Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/business/RelationshipAPITest.java:60]: Codacy found an issue: [Avoid using redundant field initializer for 'contentTypeFieldAPI'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)

**Pull request merged**